### PR TITLE
f-button@v1.0.0 - Extend f-button to include icon

### PR DESCRIPTION
**Pull request by nehadhanuka-justeat**
Add isIcon prop to f-button

- [ ] README and/or UI Documentation has been [created|updated]
- [ ] Unit tests have been [created|updated]
- [ ] This code has been checked with regard to [our accessibility standards](http://fozzie.just-eat.com/documentation/general/accessibility/checklist)

## Browsers Tested

- [ ] Chrome (latest)
- [ ] Internet Explorer 11
- [ ] Mobile (Please list device/browser – Ideally one iPhone model and one Android)

### List any other browsers that this PR has been tested in:

- [View the Browser Support Checklist](http://fozzie.just-eat.com/documentation/general/browser-support)

### CODE REVIEW

File: packages/components/atoms/f-button/CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,14 @@
 The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
 and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
 
+v0.8.0
+------------------------------
+*February 8, 2021*
+
+### Added
+- Added `isIcon` prop to the f-button
+- Added styles to display the icons properly if button is ddisplayed as an icon
```
Comment (ashleynolan) [inline packages/components/atoms/f-button/CHANGELOG.md]:
```suggestion
- Added styles to display the icons properly if button is displayed as an icon
```

File: packages/components/atoms/f-button/src/components/Button.vue [hunk 2]
```diff
@@ -4,8 +4,10 @@
         :class="[
             $style['o-btn'],
             $style[`o-btn--${buttonType}`],
-            $style[`o-btn--size${buttonSizeClassname}`],
-            (isFullWidth ? $style['o-btn--fullWidth'] : '')
+            (isIcon ? $style[`o-icon-size${buttonSizeClassname}`] :
+                $style[`o-btn--size${buttonSizeClassname}`]),
+            (isFullWidth ? $style['o-btn--fullWidth'] : ''),
+            (isIcon ? $style['o-icon'] : '')
```
Comment (ashleynolan) [inline packages/components/atoms/f-button/src/components/Button.vue:10]: I would change this to be a modifier of the base button component. Then you can chain selectors for the size as well (I'll put that comment below to show how I think that would work)

```suggestion
            (isIcon ? $style['o-btn--icon'] : '')
```

File: packages/components/atoms/f-button/src/components/Button.vue [hunk 3]
```diff
@@ -4,8 +4,10 @@
         :class="[
             $style['o-btn'],
             $style[`o-btn--${buttonType}`],
-            $style[`o-btn--size${buttonSizeClassname}`],
-            (isFullWidth ? $style['o-btn--fullWidth'] : '')
+            (isIcon ? $style[`o-icon-size${buttonSizeClassname}`] :
+                $style[`o-btn--size${buttonSizeClassname}`]),
```
Comment (ashleynolan) [inline packages/components/atoms/f-button/src/components/Button.vue:8]: I would leave this as it was, and just use the modifier to change the width of the btn 

```suggestion
            $style[`o-btn--size${buttonSizeClassname}`],
```

File: packages/components/atoms/f-button/src/components/Button.vue [hunk 4]
```diff
@@ -361,6 +376,34 @@ $btn-sizeXSmall-lineHeight      : 1;
     line-height: $btn-sizeXSmall-lineHeight;
 }
 
+/**
+* ==========================================================================
+* Icon Size Modifiers
+* ==========================================================================
+*/
+
+.o-icon {
```
Comment (ashleynolan) [inline packages/components/atoms/f-button/src/components/Button.vue:385]:
```suggestion
.o-btn--icon {
```

File: packages/components/atoms/f-button/src/components/Button.vue [hunk 5]
```diff
@@ -361,6 +376,34 @@ $btn-sizeXSmall-lineHeight      : 1;
...
+    }
+}
+
+.o-icon--sizeSmall {
+    width: $icon-sizeSmall-width;
+}
+
+.o-icon-sizeLarge {
+    width: $icon-sizeLarge-width;
+}
```
Comment (ashleynolan) [inline packages/components/atoms/f-button/src/components/Button.vue:406]: If you chain the modifiers then you can just do this:

```suggestion

     &.c-btn--sizeSmall {
        width: $icon-sizeSmall-width;
    }

    &.o-btn-sizeLarge {
        width: $icon-sizeLarge-width;
    }
}
```

**Review by ashleynolan — CHANGES_REQUESTED** [review]

**Review by dandel10n — CHANGES_REQUESTED** [review]
Could you please add an example where this is needed. At the moment we already use  buttonType='icon' property for buttons like close button with only cross icon, arrow button etc

**Comment by nehadhanuka-justeat** [comment]
> Could you please add an example where this is needed. At the moment we already use buttonType='icon' property for buttons like close button with only cross icon, arrow button etc

We want some icon buttons (like MoreVertical icon) in the new page being developed for RestaurantPartners for PromotedPlacement(https://justeattakeaway.atlassian.net/browse/PP-4835)

**Comment by ashleynolan** [comment]
@dandel10n This change will make the icon button more inline with the PIE designs, as the icon button needs to be able to use the different types of button available (primary/secondary/outline and ghost) and the way it's implemented atm this isn't possible. I'll be putting in a PR to remove the old `buttonType="icon"` once this change gets merged and we'll look to update this in any components currently using this method.

**Comment by nehadhanuka-justeat** [comment]
closing this one as again having issues with the branch from the forked repository - it is not updating the PR with the commits on the branch so have made another [PR](https://github.com/justeat/fozzie-components/pull/702) without using forked repo branch

**Pull request closed without merging**